enhancements/api-review/add-unqualified-search-registries.md

```diff
@@ -132,15 +131,20 @@ Note: adding a drop-in file at `/etc/containers/registries.conf.d` completely ov
-
-Documentations: We will document that we heavily advise against using this feature unless it is absolutely needed. An example case would be when a user has multiple internal registries whose DNS changes frequently, so image short name has to be used in the image spec. We will also document that when you do this, the whole list is overridden and there is no fall back to the default list of `unqualified-search-registries`.
+Documentations: We will document that we heavily advise against using this feature unless it is absolutely needed due to the security risks. An example case would be when a user has multiple internal registries whose DNS changes frequently, so image short name has to be used in the image spec. We will also document that when you do this, the whole list is overridden and there is no fall back to the default list of `unqualified-search-registries`. We will also document that the `unqualified-search-registries` list will not work with the builds and imagestream imports components. It will only work with the pod spec when using short names.
 
 ### Risks and Mitigations
 
-Need to ensure that all the components that deal with image management are updated to handle this new option so that they are all on the same page. The `unqualified-search-registries` field already exists in the `/etc/containers/registries.conf` file on all nodes in a cluster. This enhancement just adds the ability for the user to be able to configure this list. An API change for adding the `UnqualifiedSearchRegistry` field to the images.config.openshift.io CR is needed.
-Builds, Imagestream imports, and image-registry-pull-through will have to handle `unqualified-search-registries` as well to ensure that we are consistent across all the components that deal with image pull and management.
-Also, CRI-O will be doing the image search and will have access to the credentials for all the registries in the list.
+The `unqualified-search-registries` field already exists in the `/etc/containers/registries.conf` file on all nodes in a cluster. This enhancement just adds the ability for the user to be able to configure this list. An API change for adding the `UnqualifiedSearchRegistry` field to the images.config.openshift.io CR is needed and updates to the MCO code to handle this new option will be needed..
+CRI-O will be doing the image search and will be using the cluster wide pull secret to get the credentials for all the registries in the list.
+
+We will document a big warning about the security risks of using short names. This includes the following:
+
+- Users are subject to network and reistry-originating attacks when using external registries.
+- If the pod pull secret contains docker.io credentials, these credentials will be sent to all the registries in the search list.
+- Only the cluster-wide pull secret can be used for credentials.
```

Comment:
(plus the namespace'd docker.io creds, if present)

Reply:
Fixed.